llama_assistant/processing_thread.py

```diff
@@ -9,21 +10,31 @@ class ProcessingThread(QThread):
     update_signal = pyqtSignal(str)
     finished_signal = pyqtSignal()
 
-    def __init__(self, model, prompt, image=None):
+    def __init__(self, model, prompt, lookup_files=set(), image=None):
```

Comment:
Using set() as a default parameter value can be dangerous because it creates a mutable default argument, which is a common Python pitfall. The same set object will be shared across all instances of the class. Instead, use None and create the set inside the method:
```python
def __init__(self, model, prompt, lookup_files=None, image=None):
    self.lookup_files = set() if lookup_files is None else lookup_files
```